### PR TITLE
BUG FIX: wide_to_long modifies stubnames

### LINKED ISSUE #9204: BUG: wide_to_long modifies stubnames
This is not a serious issue, but the list of stubnames passed into `wide_to_long` is modified by the function call. 

INSTALLED VERSIONS
------------------
commit: None
python: 2.7.6.final.0
python-bits: 64
OS: Linux
OS-release: 3.13.0-43-generic
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8

pandas: 0.15.1
nose: 1.3.1
Cython: None
numpy: 1.9.1
scipy: 0.14.0
statsmodels: 0.5.0
IPython: 1.2.1
sphinx: 1.2.2
patsy: 0.2.1
dateutil: 2.3
pytz: 2014.10
bottleneck: None
tables: None
numexpr: 2.2.2
matplotlib: 1.3.1
openpyxl: 1.7.0
xlrd: 0.9.2
xlwt: 0.7.5
xlsxwriter: None
lxml: 3.3.3
bs4: 4.2.1
html5lib: 0.999
httplib2: 0.8
apiclient: None
rpy2: 2.5.2
sqlalchemy: None
pymysql: None
psycopg2: None

```python

>>> import pandas as pd
>>> df = pd.DataFrame([[0,1,2,3,8],[4,5,6,7,9]])
>>> df.columns = ['id', 'inc1', 'inc2', 'edu1', 'edu2']
>>> df
   id  inc1  inc2  edu1  edu2
0   0     1     2     3     8
1   4     5     6     7     9
>>> stubs = ['inc', 'edu']
>>> df_long = pd.wide_to_long(df, stubs, i='id', j='age')
>>> df_long
        inc  edu
id age          
0  1      1    3
4  1      5    7
0  2      2    8
4  2      6    9
>>> stubs
['edu']
```

### PR DESCRIPTION
**Pull request by jtorcasso**
Simple fix that prevents modification of the list `stubnames`, which is an argument to `wide_to_long`. 

Closes #9204

### CODE REVIEW

**Comment by shoyer** [comment]
This needs a test and an addition to the bug fixes section of what's new for 0.16.

**Comment by jtorcasso** [comment]
I have never written a test. I see you have a test written already for wide_to_long:

```python
class TestWideToLong(tm.TestCase):
    def test_simple(self):
        np.random.seed(123)
        x = np.random.randn(3)
        df = pd.DataFrame({"A1970" : {0 : "a", 1 : "b", 2 : "c"},
                           "A1980" : {0 : "d", 1 : "e", 2 : "f"},
                           "B1970" : {0 : 2.5, 1 : 1.2, 2 : .7},
                           "B1980" : {0 : 3.2, 1 : 1.3, 2 : .1},
                           "X"     : dict(zip(range(3), x))
                          })
        df["id"] = df.index
        exp_data = {"X" : x.tolist() + x.tolist(),
                    "A" : ['a', 'b', 'c', 'd', 'e', 'f'],
                    "B" : [2.5, 1.2, 0.7, 3.2, 1.3, 0.1],
                    "year" : [1970, 1970, 1970, 1980, 1980, 1980],
                    "id" : [0, 1, 2, 0, 1, 2]}
        exp_frame = DataFrame(exp_data)
        exp_frame = exp_frame.set_index(['id', 'year'])[["X", "A", "B"]]
        long_frame = wide_to_long(df, ["A", "B"], i="id", j="year")
        tm.assert_frame_equal(long_frame, exp_frame)
```

Could I just build an assertion within this existing test where I check the stubnames before and after the function call?

And how do I add to the bug fixees section? I do not think I saw in the developer's guide for contributing to pandas.

**Comment by jreback** [comment]
@jtorcasso ideally just add another tests just like this one, but that asserts the stub names don't change (you could modify this one as well, whatever works). Then add in doc/source/whatsnew/0.16.0.txt a note

**Comment by jorisvandenbossche** [comment]
@jtorcasso to be concrete, you can add another `test_...` function (like `test_simple`) to the `TestWideToLong` class. 
You can eg just use the example you showed in the issue (#9204) that failed, asserting that the stubs list in the end is still the same.

**Comment by jtorcasso** [comment]
Hopefully this last commit did the job.

File: pandas/tests/test_reshape.py [hunk 1]
```diff
@@ -443,6 +443,13 @@ def test_simple(self):
         long_frame = wide_to_long(df, ["A", "B"], i="id", j="year")
         tm.assert_frame_equal(long_frame, exp_frame)
 
+    def test_stubs(self):
+        df = pd.DataFrame([[0,1,2,3,8],[4,5,6,7,9]])
```
Comment (jorisvandenbossche) [inline pandas/tests/test_reshape.py]: Can you add here the issue number (for which this test was created), so just add `GH9204` as a comment

**Comment by jreback** [comment]
merged via https://github.com/pydata/pandas/commit/fda50121453f76142b00ff57b017b8a3ef692f69

thanks!

**Pull request closed without merging**